pkg/csi_driver/multishare_controller.go

```diff
@@ -621,13 +639,18 @@ func generateNewShare(name string, parent *file.MultishareInstance, req *csi.Cre
 	if err != nil {
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
-	return &file.Share{
+
+	share := &file.Share{
 		Name:           name,
 		Parent:         parent,
 		CapacityBytes:  targetSizeBytes,
 		Labels:         extractShareLabels(req.Parameters),
 		MountPointName: name,
-	}, nil
+	}
+	if sourceSnapshotId != "" {
```

Comment:
Actually given that "" is the nil value for a string, I may be able to get rid of the if statement and just add it in directly

Reply:
agreed, probably direct add to share object initiation would work.

Reply:
Done

Reply:
don't see this reflected in the code.. have you pushed the right changes